Source/VMware.vSphereDSC/DSCResources/VMHostSSDCache.ps1

```diff
@@ -0,0 +1,184 @@
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Datastore for Host Cache Configuration from the server if it exists.
+    If the Datastore does not exist, it throws an exception.
+    #>
+    [PSObject] GetDatastore() {
+        try {
+            $foundDatastore = Get-Datastore -Server $this.Connection -Name $this.Datastore -ErrorAction Stop
+            return $foundDatastore
+        }
+        catch {
+            throw "Could not retrieve Datastore $($this.Datastore) for VMHost $($this.Name). For more information: $($_.Exception.Message)"
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Cache Info for the specified Datastore from the Host Cache Configuration.
+    If the Datastore is not enabled for swap performance, it throws an exception.
+    #>
+    [PSObject] GetDatastoreCacheInfo($vmHostCacheConfigurationManager, $foundDatastore) {
+        $datastoreCacheInfo = $vmHostCacheConfigurationManager.CacheConfigurationInfo | Where-Object { $_.Key -eq $foundDatastore.ExtensionData.MoRef }
+        if ($null -eq $datastoreCacheInfo) {
+            throw "Datastore $($foundDatastore.Name) could not be found in enabled for swap performance Datastores."
+        }
+
+        return $datastoreCacheInfo
+    }
+
+    <#
+    .DESCRIPTION
+
+    Converts the passed MB value to GB value by rounding it down.
+    #>
+    [int] ConvertMBValueToGBValue($mbValue) {
+        $megabytesInGB = 1024
+        return [int] [Math]::Floor($mbValue * $megabytesInGB * $megabytesInGB / 1GB)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Checks if the Host Cache Configuration should be updated for the specified VMHost by checking
+    if the current Swap Size is equal to the desired one for the specified Datastore.
+    #>
+    [bool] ShouldUpdateHostCacheConfiguration($vmHostCacheConfigurationManager) {
+        $foundDatastore = $this.GetDatastore()
+        $datastoreCacheInfo = $this.GetDatastoreCacheInfo($vmHostCacheConfigurationManager, $foundDatastore)
+
+        return ($this.ConvertMBValueToGBValue($this.SwapSize) -ne $this.ConvertMBValueToGBValue($datastoreCacheInfo.SwapSize))
+    }
+
+    <#
+    .DESCRIPTION
+
+    Performs an update on the Host Cache Configuration of the specified VMHost by changing the Swap Size for the
+    specified Datastore.
+    #>
+    [void] UpdateHostCacheConfiguration($vmHostCacheConfigurationManager) {
+        $foundDatastore = $this.GetDatastore()
+
+        $hostCacheConfigurationSpec = New-Object VMware.Vim.HostCacheConfigurationSpec
+        $hostCacheConfigurationSpec.Datastore = $foundDatastore.ExtensionData.MoRef
+        $hostCacheConfigurationSpec.SwapSize = $this.SwapSize
+
+        $hostCacheConfigurationResult = Update-HostCacheConfiguration -VMHostCacheConfigurationManager $vmHostCacheConfigurationManager -Spec $hostCacheConfigurationSpec
+        $hostCacheConfigurationTaskState = $null
+        $sleepTimeInSeconds = 5
+
+        while ($true) {
```

Comment:
Why don't you use Wait-Task cmdlet instead of implementing wait for completion?

Reply:
Done.